### PR TITLE
[#4] Setup cicd

### LINKED ISSUE #4: [Chore]Setup CICD
## Why

Set up the CI and CD workflow

## Acceptance Criteria

- Deploy workflow should run successfully on separate environments which are Stagging and Production
- The application should run unit and ui test after developers create new PR to prevent any failure

### PR DESCRIPTION
**Pull request by suphanatjarukulgowit**
- Close https://github.com/suphanatjarukulgowit/react-nimble-survey/issues/4

## What happened 👀

Set up cicd with GitHub action and deploy to netlify

## Insight 📝

There's an error when the workflows action tries to deploy to netlify this happens because there's no permission to do an action.
image

There're 2 ways to solve it
1.Add permission to workflow
image
2.Set permission in setting of repository
image

## Proof Of Work 📹
The workflow is run successfully
image

image

### CODE REVIEW

**Review by bterone — DISMISSED** [review]

File: .tool-versions [hunk 1]
```diff
@@ -1 +1,2 @@
-nodejs 16.14.2
+nodejs ^18.0.0
+nvm ^0.39.0
```
Comment (bterone) [inline .tool-versions]: Minor comment, but can you end on a new line here 👍

**Review by bterone — COMMENTED** [review]
Don't forget to rebase this branch from https://github.com/suphanatjarukulgowit/react-nimble-survey/pull/30, some changes here are from that branch

File: .tool-versions [hunk 2]
```diff
@@ -1 +1,2 @@
-nodejs 16.14.2
+nodejs ^18.0.0
```
Comment (bterone) [inline .tool-versions]: I think its fine to pin the same version as the one you defined in the Prerequisites of your README file 😄

Comment (suphanatjarukulgowit) [reply .tool-versions]: updated nodejs version to 18.13.0 LTS 😄

File: .tool-versions (hunk 1 above)
Comment (suphanatjarukulgowit) [reply .tool-versions]: New line is added 👍

**Review by hoangmirs — COMMENTED** [review]
Could you rebase this PR against `chore/3-setup-ui-testing`

**Review by hoangmirs — APPROVED** [review]

**Review by bterone — APPROVED** [review]

**Pull request merged**